packages/react-sdk/src/client.ts

```diff
@@ -0,0 +1,595 @@
+/**
+ * Copyright 2019, Optimizely
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as optimizely from '@optimizely/optimizely-sdk'
+import * as logging from '@optimizely/js-sdk-logging'
+
+const logger = logging.getLogger('ReactSDK')
+
+export type VariableValuesObject = {
+  [key: string]: boolean | number | string | null
+}
+
+export type UserAttributes = { [attribute: string]: any }
+
+type EventTags = { [tagKey: string]: boolean | number | string }
+
+type DisposeFn = () => void
+
+type OnUserUpdateHandler = (userInfo: UserInfo) => void
+
+export type OnReadyResult = {
+  success: boolean
+  reason?: string
+}
+
+export interface ReactSDKClient extends optimizely.Client {
```

Comment:
It is sooo weird that an interface also extends another class. Shouldn't the `OptimizelyReactSDKClient` be the one extending `optimizely.Client`? Also, the `OptimizelyReactSDKClient` seems to wrap an instance of the optly client, so what is the purpose of the inheritance here?

Reply:
`optimizely.Client` is an interface, not a class. This is defining the interface `ReactSDKClient` which extends `optimizely.Client`. There is no inheritance.

The class that implements this interface is `OptimizelyReactSDKClient` defined below. Jordan likes to define the interface and the class that implements it separately even when no other classes uses the interface. I don't see much of a point in this, so would be willing to change if you don't like it.

Reply:
Actually, I think we can remove the `extends`. Everything is being overridden anyway. Thanks for the tip.

Reply:
Makes sense to me. I think it is fine here.